CalibPPS/ESProducers/plugins/CTPPSBeamParametersFromLHCInfoESSource.cc

```diff
@@ -31,16 +31,23 @@ class CTPPSBeamParametersFromLHCInfoESSource : public edm::ESProducer {
   static void fillDescriptions(edm::ConfigurationDescriptions&);
 
 private:
-  const edm::ESGetToken<LHCInfo, LHCInfoRcd> lhcInfoToken_;
+  edm::ESGetToken<LHCInfo, LHCInfoRcd> lhcInfoToken_;
+  edm::ESGetToken<LHCInfoPerLS, LHCInfoPerLSRcd> lhcInfoPerLSToken_;
+  edm::ESGetToken<LHCInfoPerFill, LHCInfoPerFillRcd> lhcInfoPerFillToken_;
+  const bool useNewLHCInfo_;
 
   CTPPSBeamParameters defaultParameters_;
 };
 
 //----------------------------------------------------------------------------------------------------
 
 CTPPSBeamParametersFromLHCInfoESSource::CTPPSBeamParametersFromLHCInfoESSource(const edm::ParameterSet& iConfig)
-    : lhcInfoToken_(
-          setWhatProduced(this).consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoLabel")))) {
+    : useNewLHCInfo_(iConfig.getParameter<bool>("useNewLHCInfo")) {
+  auto cc = setWhatProduced(this);
+  lhcInfoToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoLabel")));
+  lhcInfoPerLSToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoPerLSLabel")));
+  lhcInfoPerFillToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoPerFillLabel")));
```

Comment:
This does not sound like a bad idea, however in every module I worked on `lhcInfoLabel` is a string instead of `ESInputTag` so I think there might be a reason for it. It's worth investigating but I would defer it to a future PR - it's an aesthetic update, needed in many files and will take a while to implement and test

Reply:
> It's worth investigating but I would defer it to a future PR

That would be be perfectly fine, thanks.

(in principle using `ESInputTag` directly in configuration opens the possibility to use the first "module label" field of `ESInputTag` for ensuring the ES data products come from the expected ES module, if such requirement would be wanted, but that feature is not really used in CMSSW application configuration)